src/main/java/net/snowflake/client/core/BaseSqlInput.java

```diff
@@ -103,4 +114,53 @@ public SQLXML readSQLXML() throws SQLException {
   public RowId readRowId() throws SQLException {
     throw new SnowflakeLoggedFeatureNotSupportedException(session, "readRowId");
   }
+
+  protected <T> T readObjectOfType(Class<T> type, Object value, SQLInput sqlInput)
+      throws SQLException {
+    if (SQLData.class.isAssignableFrom(type)) {
+      SQLData instance = (SQLData) SQLDataCreationHelper.create(type);
+      if (sqlInput == null) {
+        return null;
+      } else {
+        instance.readSQL(sqlInput, null);
+        return (T) instance;
+      }
+    } else if (Map.class.isAssignableFrom(type)) {
+      Object object = value;
+      if (object instanceof JsonSqlInput) {
```

Comment:
since this is in abstract class already, we can replace it with some private abstract method implemented in both subclasses.

Reply:
Fixed